packages/cozy-mespapiers-lib/src/components/Search/helpers.js

```diff
@@ -0,0 +1,40 @@
+import { FILES_DOCTYPE } from '../../doctypes'
+import { index, addDocs } from './search'
+
+const updateFile = doc => {
+  if (doc.trashed) {
+    index.remove(doc._id)
+  }
+}
+
+export const onUpdate = doctype => async doc => {
+  if (doctype === FILES_DOCTYPE) {
+    return updateFile(doc)
+  }
+}
+
+export const onCreate = (doctype, scannerT) => async doc => {
+  addDocs({ index, docs: [{ ...doc, _type: doctype }], scannerT })
+}
+
+export const add = (isInit, scannerT, setIsInit) => docs => {
+  if (!isInit) {
+    addDocs({ index, docs, scannerT })
+    setIsInit(true)
+  }
```

Comment:
nit: If the indexing is only for possible for init, why not naming this something like `initialDocsIndexing`? It feels a bit weird to have an `add` method that:
- add multiple docs
- only works if init is false